roxie/udplib/udpsha.hpp

```diff
@@ -176,70 +174,52 @@ class simple_queue
 #define HANDLE_PRAGMA_PACK_PUSH_POP
 #endif
 
-#pragma pack(push,1)
-struct UdpPermitToSendMsg
-{
-    // New static fields must be included inside this block, so that
-    // size calculations work correctly
-    unsigned short  length;
-    unsigned short  cmd;
-    unsigned short  destNodeIndex;
-    unsigned short  max_data;
-#ifdef CRC_MESSAGES
-    unsigned        crc;
-#endif
-
-#ifdef CRC_MESSAGES
-    unsigned calcCRC()
+class flow_t {
+public:
+    enum flowmsg_t  { ok_to_send, request_received, request_to_send, send_completed, request_to_send_more };
```

Comment:
Should explicitly define the base type ( e.g. :short) because it is included in a structure that is treated as POD and written to a socket.

Reply:
I thought I had (must have got lost while fighting weird initializer issues)